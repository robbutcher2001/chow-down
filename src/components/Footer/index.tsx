import React from 'react';

import styled from 'styled-components';

import { RawLink } from '../Clickable';

const Footer = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-size: 0.8rem;
  padding: 3rem 0;
  color: #c0c0c0;
  background-color: #fafafa;

  > p {
    margin: 1rem;
    text-align: center;
  }
`

export default () => (
  <Footer>
    <p>&copy; Rob Butcher 2020.</p>
    <p>
      {'Images: '}
      <RawLink
        $inline
        $underline
        $small
        href='https://unsplash.com/'
        target='_blank'>
        Unsplash
      </RawLink>
      {' and various recipe websites.'}
    </p>
  </Footer>
);