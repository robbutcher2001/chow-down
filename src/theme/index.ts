import { DefaultTheme } from 'styled-components';

import colour from './defs/_colour';

const theme: DefaultTheme = {
  colour
};

export default theme;