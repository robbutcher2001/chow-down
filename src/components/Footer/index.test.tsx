import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../../theme';
import Footer from '.';

test('Footer basic snapshot render', () => {
  const footer = renderer.create(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>
  );

  expect(footer.toJSON()).toMatchSnapshot();
});

test('Footer correct tag content assertion', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>
  );

  expect(getByText(/Rob Butcher/).textContent).toEqual('Â© Rob Butcher 2020.');
  expect(getByText(/Images/).textContent).toEqual('Images: Unsplash and various recipe websites.');
});