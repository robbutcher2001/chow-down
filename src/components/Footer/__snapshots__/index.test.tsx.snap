// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer basic snapshot render 1`] = `
<footer
  className="sc-AxhUy hgSSXO"
>
  <p>
    Â© Rob Butcher 2020.
  </p>
  <p>
    Images: 
    <a
      className="sc-AxiKw kSEELl"
      href="https://unsplash.com/"
      target="_blank"
    >
      Unsplash
    </a>
     and various recipe websites.
  </p>
</footer>
`;
